lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -934,7 +934,7 @@ export class UserAgentApplication {
       let authErr: AuthError;
       let cacheResultResponse;
 
-      if (!userContainedClaims) {
+      if (!userContainedClaims && !request.forceRefresh) {
```

Comment:
Any possibility for the request to be null?  login() calls -> acquireTokenSilent() is a path that can have a null request, are we handling this? I hope we did but please cross check.

Reply:
ATS on it's own checks params in request above this , so if it is broken, I do not believe it is by this

Reply:
We require a request object for ATS calls, it is not an optional parameter.

Reply:
Actually this is a bug, since we don't null check the object. May want to add this.

Reply:
It is not clear to me what the right behavior is here.
If a request object is not provided, should we throw from ATS?

Reply:
You just need to check against request being null. The code flow here even if request is null is when ats is called for adal idtoken cases

Reply:
Mostly to avoid referencing undefined objects

Reply:
What i mean is if you look at the code above it, there are lots of accessors to the the request object, so if it is null, it should break anyway right ?

Reply:
updating to fix my code

Reply:
Do we have consumers who call ATS without passing in a request object?

Reply:
Because looking at the code, that would not be supported.

Reply:
That does not appear to be supported, and in terms of code readability we shouldn't added null checks just because it could be null if its not actually supported. If a null request object isn't supported, we should probably add an assertion at the beginning of the method. If it is supposed to be supported, there are more checks that need to be added (and probably some documentation).

Reply:
Yes. That was the intent of the original comment, it is to ensure we have the code guarded against null request to avoid undefined references. It need not necessarily be a null check and as you said, a generic assertion suffices and is a better approach.